sonic-slave-bullseye/Dockerfile.j2

```diff
@@ -44,11 +52,68 @@ RUN echo "deb [arch=arm64] http://deb.debian.org/debian bullseye main contrib no
         echo 'deb [arch=arm64] http://ftp.debian.org/debian bullseye-backports main' >> /etc/apt/sources.list && \
         echo "deb [arch=arm64] http://packages.trafficmanager.net/debian/debian bullseye main contrib non-free" >> /etc/apt/sources.list && \
         echo "deb [arch=arm64] http://packages.trafficmanager.net/debian/debian bullseye-updates main contrib non-free" >> /etc/apt/sources.list
+{%- elif CROSS_BUILD_ENVIRON == "y" %}
+{%- if CONFIGURED_ARCH == "armhf" %}
+ARG arch=armhf
+ARG gcc_arch=arm-linux-gnueabihf
+ARG PYTHON_CROSS_PLATFORM=linux_armv7l
+{%- elif CONFIGURED_ARCH == "arm64" %}
+ARG arch=arm64
+ARG gcc_arch=aarch64-linux-gnu
+ARG PYTHON_CROSS_PLATFORM=linux_aarch64
+{%- endif %}
+
+RUN dpkg --add-architecture $arch
+RUN apt-get update
+RUN apt-get install -y crossbuild-essential-$arch
+RUN apt-get install -y gcc-$gcc_arch
+
+RUN apt-mark hold g++-$gcc_arch
+RUN apt-mark hold g++-10-$gcc_arch
+RUN apt-mark hold gcc-$gcc_arch
+RUN apt-mark hold gcc-10-$gcc_arch
+
+ARG CROSS_CC=${gcc_arch}-gcc
+ARG CROSS_CXX=${gcc_arch}-g++
 {%- endif %}
 
 ## Make apt-get non-interactive
 ENV DEBIAN_FRONTEND=noninteractive
 
+{%- if CROSS_BUILD_ENVIRON == "y" %}
+RUN apt-get install -y python3 python3-pip
+RUN apt-get install -y python3-minimal:$arch python3.9:$arch python3:$arch python3-dev:$arch python3-setuptools:$arch 
+RUN apt-get download python3-distutils && dpkg --force-all -i python3-distutils*
+RUN apt-get download python3-pip && dpkg --force-all -i python3-pip*
+RUN which pip3 && pip3 install enum34
+RUN pip3 install virtualenv
+
+# Create target arm python3 virtual environments with all required packages installed
+RUN mkdir /python_virtualenv
+RUN cd /python_virtualenv && python3 -m virtualenv --copies -p /usr/bin/python3 env3
+
+RUN PATH=/python_virtualenv/env3/bin/:$PATH pip3 install setuptools==49.6.00 wheel==0.35.1 fastentrypoints pytest pytest-cov pytest-runner==5.2 nose==1.3.7 mockredispy==2.9.3 mock==3.0.5 PyYAML==5.4.1 redis==3.5.3 pexpect==4.8.0 Pympler==0.8 parameterized natsort==6.2.1 MarkupSafe==2.0.1 Jinja2==3.0.3 click tabulate netaddr netifaces pyroute2 pyfakefs sphinx && ln -s /python_virtualenv/env3/bin/pytest /python_virtualenv/env3/bin/pytest-3 
+
+RUN apt-get --fix-broken install -y
+RUN LIBPYTHON3_DEPS="`apt-cache depends libpython3-dev:$arch |grep Depends|awk {'print \$2;'}|tr "\n" " "`" && apt-get install -y libpython2.7-dev:$arch $LIBPYTHON3_DEPS libxml2-dev:$arch libxslt-dev:$arch libssl-dev:$arch libz-dev:$arch
+
+RUN apt-get install -y swig libssl-dev
+
+# Cross-compile the heaviest python packages for arm
+ARG pre_built_target_dir=PRE_BUILT_TARGET 
+RUN rm -rf $pre_built_target_dir && mkdir $pre_built_target_dir && rm -rf python-cache && mkdir python-cache
+RUN PATH=/python_virtualenv/env3/bin/:$PATH CC=$CROSS_CC pip3 install --cache-dir python-cache grpcio==1.39.0 grpcio-tools==1.20.0 m2crypto==0.36.0 bitarray==1.5.3 lxml==4.6.5
```

Comment:
These packages (like grpcio) is part of sonic image. They should not be built within slave image. They are similar to linux-kernel-header packages, which are built inside a slave container.